### DISCUSSION TITLE
Issue with SortBinaryRegion

### DISCUSSION THREAD

**Discussion by AugustoAnesio** [bonsai-rx/bonsai#1474] [Q&A]
Hello, I am using Bonsai to track two rats in an open field. One of the is trapped in a wired cage, while the other can circulate around. 
I used BinaryRegionAnalysis and SortBinaryRegions node thinking that I would have the isolated centroid from both animals (to calculate distance, trajectory, etc). However, BinaryRegionAnalysis and SortBinaryRegions sort each collection of binary regions in the sequence **by descending order of area.**. As both rats have similar sizes, the centroid coordinates get mixed, and I haven't figured out how to isolate them. 
One of my ideas is to create a trajectory map of the rat going around, but because of this issue, I still having this driftings on centroid (image below)

Anyone can help?

Thank you 

![image](https://github.com/bonsai-rx/bonsai/assets/118387579/1611322a-d6cc-419f-a254-cb5b1b649dbf)

![image](https://github.com/bonsai-rx/bonsai/assets/118387579/74e55de9-7963-4c69-8f9e-f497ef462648)

![image](https://github.com/bonsai-rx/bonsai/assets/118387579/97f8d0d5-7f92-4524-a489-2f95ad94720d)

**Comment by bruno-f-cruz** [comment]
Just to clarify, is it the case that you just want to keep the identity of each animal constant? Wouldnt simply take the top 2 areas and use a simple heuristic, e.g. If centroid in cage => animal 1, else => animal2 work? I assume the cage doesn't move, so this should work?

**Reply by AugustoAnesio** [reply]
Hello Bruno, thank you for your answer.

Sorry, but I did not understand your suggestion.  

The cage doesn't move. But the areas do vary a lot because the position changes. And yes, I want to keep the identity

**Reply by bruno-f-cruz** [reply]
I am suggesting you use the position of the cage as a proxy for the identity of the animal inside the cage. Once you know which animal is inside the box, the identity of the other animal can also be inferred. Does this make sense?

**Reply by AugustoAnesio** [reply]
Got it, like defining that each coordinate inside cage limits are from animal1 and outside from animal2, then separate them based on this rule? 

Thank you!

**Reply by bruno-f-cruz** [reply]
yes!
